### ISSUE TITLE
Thoughts on merging axlsx_styler into caxlsx directly

### ISSUE THREAD

**Issue by westonganger** [caxlsx/caxlsx#167]
It seems to me that the additional features offered in [`axlsx_styler`](https://github.com/axlsx-styler-gem/axlsx_styler) are a very nice upgrade to `caxlsx`. I have been using it successfully with `spreadsheet_architect` for over 6 years. Now that axlsx is actually maintained properly, Would the team consider merging `axlsx_styler` into `caxlsx` directly? 

I am more than happy to keep maintaining `axlsx_styler` on its own however I had the thought that the overall community might be better served by adding this. Most important feature provided by `axlsx_styler` is the ability to apply styles after row creation (while still maintaining the previous styles applied)

If we were to move forward with this. I would be happy to be full contributor to `caxlsx`. You would likely be familiar with me as I have provided many enhancements to `caxlsx` since the split from `axlsx`.

**Comment by kiskoza** [comment]
I don't see an reason not to do it, the DSL of axlsx_styler looks very decent to me.

However, I think we have several details we need to figure out:

1. I've never done gem merging before - can it cause any troubles when you have both the future version of caxlsx and axslx_styler in your gemfile? Of course, we should release a new version for axlsx_styler with a deprecation message, but it would be nice to somehow detect the existence of the styler gem from caxlsx too, at least in the version where we merge them and show a deprecation message
2. It would be nice to make a clean release for this one where we don't have any other bug fixes / code changes apart the merge itself. In my opinion it can be a minor version, it shouldn't have any breaking changes
3. We need to bring over all the existing examples and make sure we update caxlsx's examples when something is easier with the new DSL
4. Need to double check the whole codebase to not bring over deprecated methods / ways of styling. Also need to check dependencies, etc to keep caxlsx in a compact state

**Comment by westonganger** [comment]
#1 would be easy. Just add code within the new caxlsx versions for 

```ruby
if defined?(AxlsxStyler::VERSION)
  raise "Please remove axlsx_styler from your Gemfile"
end
```

**Comment by westonganger** [comment]
@noniq @straydogstudio any thoughts?

**Comment by noniq** [comment]
In general I’m a fan of keeping things separate. But as far as I understand here we’re talking about adding a rather small amount of code for getting a big benefit in usability. I like the idea!

**Comment by straydogstudio** [comment]
That sounds like a good idea to me. We have had several questions on easier styling. Kiskoza: all of your points are valid and I can’t think of anything else. Keeping reduced dependencies is important. 

It could be a major release. I know it isn’t a breaking change. But it is a significant addition and has the potential to cause problems through code or dependencies (which we will try to iron out). It would be a way of telling that to the community. I’m fine either way however. 

I suggest, if we merge or not, moving the existing gem as is into the caxlsx organization. Is that desirable? For use by those using older versions of caxlsx or axlsx?

**Comment by westonganger** [comment]
Ok sounded like general buy-in here. I went ahead and made a quick first pass at this in #168. Lets work together to get this across the finish line, reviews, improvements & suggestions wanted.

**Comment by westonganger** [comment]
In regards to moving `axlsx_styler` into the github org.

If we merge this functionality, I think it may be slightly confusing to other new users if we add this repo under the namespace. Then again should be straight forward given the post-merge steps given in #168. I dont know, I think that I am not sure what the gain would be here.

If we dont merge, then likely this is a good idea.

**Comment by straydogstudio** [comment]
I see value in placing a copy of the axlsx_styler source code (in its current state) in the Caxlsx community. This would be for historical purposes, so the source code doesn't get lost. There's a small risk that your copy would be removed, but it is a possibility. The README could clearly indicate it is deprecated. I've lost code before, so I try to prepare for it. This is, however, a precaution, and not necessary. 

In my experience most people don't explore other repos that belong to a given author. But some will stumble across it. Unless you guys see value in it, I'll drop the suggestion.

**Comment by westonganger** [comment]
This has now been merged into caxlsx master and released in v3.3.0.
